src/collection-view.js

```diff
@@ -641,22 +643,29 @@ const CollectionView = Backbone.View.extend({
     return this;
   },
 
+  addChildViews(views, opts = {}) {
+    if(!_.isArray(view)) return;
+    
+    _.each(views, (view) => this.addChildView(view, opts.index, false));
+    
+    if(this.isRendered() && opts.shouldTriggerShow !== false)
+      this._showChildren();
+  }
   // Render the child's view and add it to the HTML for the collection view at a given index, based on the current sort
-  addChildView(view, index) {
+  addChildView(view, index, shouldTriggerShow = true) {
     if (!view || view._isDestroyed) {
       return view;
     }
 
-    if (!this._isRendered) {
-      this.render();
-    }
-
     // Only cache views if added to the end
-    if (!index || index >= this.children.length) {
-      this._addedViews = [view];
+    if (index == null || index >= this.children.length) {
+      this._addedViews || (this._addedViews = []);
```

Comment:
I think if `shouldTriggerShow` is introduced the entire `addedViews` perf additional will need to be removed.  (because other areas do not also add to an existing array of views that have been added but not shown)  I think it will introduce caveats all over the place to not have a standard logic flow.  Possibly elsewhere as well.